browser/ui/webui/brave_rewards_internals_ui.cc

```diff
@@ -401,6 +413,46 @@ void RewardsInternalsDOMHandler::OnGetEventLogs(ledger::type::EventLogs logs) {
       std::move(data));
 }
 
+void RewardsInternalsDOMHandler::GetAdDiagnostics(const base::ListValue* args) {
+  if (!ads_service_) {
+    NOTREACHED();
+    return;
+  }
+
+  ads_service_->GetAdDiagnostics(
+      base::BindOnce(&RewardsInternalsDOMHandler::OnGetAdDiagnostics,
+                     weak_ptr_factory_.GetWeakPtr()));
+}
+
+void RewardsInternalsDOMHandler::OnGetAdDiagnostics(const bool success,
+                                                    const std::string& json) {
+  if (!web_ui()->CanCallJavascript()) {
+    return;
+  }
+
+  base::Value diagnostics(base::Value::Type::LIST);
+  if (success && !json.empty()) {
+    absl::optional<base::Value> serialized_json = base::JSONReader::Read(json);
+    if (serialized_json && serialized_json->is_list() &&
+        !serialized_json->GetList().empty()) {
+      diagnostics = std::move(*serialized_json);
+    }
+  }
+
+#if DCHECK_IS_ON()
+  DCHECK(diagnostics.is_list()) << "Diagnostics should be a list";
+  for (const auto& entry : diagnostics.GetList()) {
+    DCHECK(entry.is_dict()) << "Diagnostics entry should be a dictionary";
+    DCHECK(entry.FindKey("key")) << "Diagnostics entry should have 'key' key";
+    DCHECK(entry.FindKey("value"))
+        << "Diagnostics entry should have 'value' key";
+  }
+#endif  // DCHECK_IS_ON()
+
+  web_ui()->CallJavascriptFunctionUnsafe(
```

Comment:
why not use https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/web_ui.h;l=101-103?q=CallJavascriptFunctionUnsafe&ss=chromium instead? seems mostly the same except this also checks `javascript_allowed_` https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/web_ui_message_handler.h;drc=9d1ccebc2fa67f5a624835e692f3f27c7e60acae;bpv=1;bpt=1;l=155

Reply:
`CallJavascriptFunctionUnsafe` call is used in a lots of places in this file. I think it would be better to fix them all in a separate ticket. Created an issue for this:
https://github.com/brave/brave-browser/issues/17455

Reply:
@aseren thanks!